piwheels/master/the_scribe.py

```diff
@@ -349,6 +361,76 @@ def write_project_page(self, package):
         except FileExistsError:
             pass
 
+    def delete_package(self, package):
+        """
+        Attempts to remove the index and project page directories (including all
+        known wheel files) of the specified *package*.
+
+        :param str package:
+            The name of the package to delete.
+        """
+        if len(package) == 0:
+            # refuse to delete /simple/ and /project/ by accident
+            self.logger.error('not deleting - package name is empty')
+            return
+
+        files = {f for f in self.db.get_package_files(package)}
+
+        for name in ('simple', 'project'):
+            self.logger.info('deleting /%s/ dir for %s', name, package)
+            pkg_dir = self.output_path / 'simple' / package
+            canon_dir = pkg_dir.with_name(canonicalize_name(pkg_dir.name))
+
+            for dir in (pkg_dir, canon_dir):
+                self.logger.debug('Processing %s', dir)
+                index = dir / 'index.html'
+                for file in files:
+                    file_path = dir / file
+                    try:
+                        file_path.unlink()
+                        self.logger.debug('File deleted: %s/%s', dir, file)
+                    except FileNotFoundError:
+                        self.logger.debug('File not found: %s/%s', dir, file)
```

Comment:
Shouldn't be debug; this is us trying to delete a file we were sure existed and it turns out it doesn't. Should be a warning.

Reply:
It might not be though! If we already deleted it via the previous attempt (symlink or direct) then you'd expect it to not find the file.

Reply:
Ohhh ... good point. Sorry, I'd (incorrectly) assumed the removal of a redundant loop would prevent that but of course it won't. Hmmm. In that case this feels ... odd to me. Ideally I want something where we never attempt a deletion we don't *think* exists (according to the database); I'll have a think about this and see if I can come up with something.

Reply:
How about we pop the file off the set once it's deleted?